File: 0001_0100/0067/0067.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    p = []
    for i in range(n):
        r = 0
        for x in (int(x) for x in f.readline().strip().split('.')):
            r = r * 256 + x
        p.append(r)
    for i in range(int(f.readline())):    
        s, t = [x.split('.') for x in f.readline().split()]
        a = b = 0
        for x in (int(x) for x in s):
            a = a * 256 + x
        for x in (int(x) for x in t):
            b = b * 256 + x
        q.write(str(sum(1 for j in p if a & j == b & j)) + "\n")           
        

File: 0001_0100/0067/0067.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func readIP(s string) uint {
	t := strings.Split(s, ".")
	a, _ := strconv.ParseUint(t[0], 0, 0)
	b, _ := strconv.ParseUint(t[1], 0, 0)
	c, _ := strconv.ParseUint(t[2], 0, 0)
	d, _ := strconv.ParseUint(t[3], 0, 0)
	return uint((((((a << 8) | b) << 8) | c) << 8) | d)
}

func main() {
	c, _ := ioutil.ReadFile("input.txt")
	q, _ := os.Create("output.txt")
	s := strings.Fields(string(c))
	n, _ := strconv.Atoi(s[0])
	p := make([]uint, n)
	for i := range p {
		p[i] = readIP(s[i+1])
	}
	m, _ := strconv.Atoi(s[n+1])
	for i := 0; i < 2*m; i += 2 {
		k := 0
		x, y := readIP(s[n+i+2]), readIP(s[n+i+3])
		for _, v := range p {
			if x&v == y&v {
				k++
			}
		}
		fmt.Fprintln(q, k)
	}
}
